### PR TITLE
add `databricks_current_config` data source

### LINKED ISSUE #2867: [FEATURE] Add a new data source to get information about current configuration - cloud type, host, etc.
Sometimes, when you write modules you may need to get information about the currently configured provider to make a decision, for example, add a dynamic block based on the specific cloud. We have [workspace_url](https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/current_user#workspace_url) attribute in the `databricks_current_user` data source, but it's specific to workspace-level operations, and doesn't work on account level.

It would be useful to add a separate data source that will expose the following attributes:

* if it's account or workspace-level 
* configured `host`
* cloud type - GCP/AWS/Azure
* authentication type (?)
* ...

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
Closes #2867

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Review by alexott — APPROVED** [review]
lgtm!

File: mws/data_current_config.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+			data.IsAccount = true
+		}
+		data.Host = c.Config.Host
+		if c.Config.IsAws() {
+			data.CloudType = "aws"
+		} else if c.Config.IsAzure() {
+			data.CloudType = "azure"
+		} else {
+			data.CloudType = "gcp"
+		}
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply mws/data_current_config.go]: updated

**Comment by nkvuong** [comment]
integration tests passed - tests failed are unrelated

**Review by mgyucht — APPROVED** [review]
This is terrific. LGTM, just can you double check that the internal test cases are actually executed in nightlies?

**Pull request merged**